### PR TITLE
[SC 4057] Add residuals plot test for regression models

### PR DESCRIPTION
**Pull request by AnilSorathiya**
## Internal Notes for Reviewers
* Add residuals plots for regression models
### Additional changes
* Fix the FeatureTargetCorrelationPlot test
* Fix the test description generation script

### CODE REVIEW

**Review by johnwalz97 — APPROVED** [review]
LGTM

File: validmind/tests/model_validation/RegressionResidualsPlot.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    metadata = {
+        "task_types": ["regression"],
+        "tags": [
+            "model_performance",
+        ],
+    }
+
+    def run(self):
+        y_true = self.inputs.dataset.y
+        y_pred = self.inputs.dataset.y_pred(self.inputs.model.input_id)
```
Comment (cachafla) [inline validmind/tests/model_validation/RegressionResidualsPlot.py:64]: With this PR merged https://github.com/validmind/developer-framework/pull/44 this code should be:

```suggestion
        y_pred = self.inputs.dataset.y_pred(self.inputs.model)
```

**Review by cachafla — APPROVED** [review]
Nice 👌. I left a change suggestion to be compatible with the new `y_pred` notation.

File: validmind/tests/model_validation/RegressionResidualsPlot.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        scatter = go.Scatter(
+            x=y_true.flatten(),
+            y=y_pred.flatten(),
+            mode="markers",
+            name="True vs Predicted",
+            marker=dict(color="blue", opacity=0.5),
+        )
+
+        # Line of perfect prediction
+        max_val = max(max(y_true), max(y_pred))
```
Comment (juanmleng) [inline validmind/tests/model_validation/RegressionResidualsPlot.py:96]: To make this a bit more robust against potential NaNs, you can use: 
 `max_val = np.nanmax([np.nanmax(y_true), np.nanmax(y_pred)])`
 `min_val = np.nanmin([np.nanmin(y_true), np.nanmin(y_pred)])`

File: validmind/tests/model_validation/RegressionResidualsPlot.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        y_true = self.inputs.dataset.y
+        y_pred = self.inputs.dataset.y_pred(self.inputs.model.input_id)
+        # Calculate residuals
+        residuals = y_true.flatten() - y_pred.flatten()
+        # Create residuals plot
+        hist_data = [residuals]
+        group_labels = ["Residuals"]  # Names of the dataset
+
+        fig = ff.create_distplot(
+            hist_data, group_labels, bin_size=[0.1], show_hist=True, show_rug=False
```
Comment (juanmleng) [inline validmind/tests/model_validation/RegressionResidualsPlot.py:72]: For histograms, it is always convenient to allow users to specify the bin size as a parameter:
`bin_size = self.params["bin_size"]`

**Review by juanmleng — APPROVED** [review]
I left some minor comments. Other than that, this LGTM :)

File: scripts/add_test_description.py [hunk 4]
```diff
@@ -186,7 +186,7 @@ def add_summary_to_test(path):
     with open(path, "r") as f:
         file_contents = f.read()
 
-    response = openai.ChatCompletion.create(
+    response = openai.chat.completions.create(
         model="gpt-4", # or gpt-3.5-turbo
```
[the remark this answers is not in the document]
Comment (AnilSorathiya) [reply scripts/add_test_description.py:190]: 👍

**Pull request merged**